src/main/java/com/github/tomakehurst/wiremock/store/files/FileSourceBlobStore.java

```diff
@@ -93,6 +93,13 @@ public void remove(String key) {
     fileSource.deleteFile(key);
   }
 
+  @Override
+  public Optional<byte[]> getAndRemove(String key) {
+    Optional<byte[]> value = get(key);
+    remove(key);
+    return value;
```

Comment:
i decided to forgo locking here, since this implementation is currently not using any locks, but i could add something similar to FileSourceJsonObjectStore pretty easily.